msal/application.py

```diff
@@ -174,6 +173,7 @@ def __init__(
                 # when we would eventually want to add this feature to PCA in future.
             exclude_scopes=None,
             http_cache=None,
+            instance_discovery=None,
```

Comment:
Is instance_discovery=None here means default to False?

I guess we want the default value to be true?

Reply:
Basically, I think I am moving away from using `True` or `False` as the default value, because they would force downstream yet mid-tier libraries (such as Azure Identity) to also duplicate the same default value in their corresponding API surface.

I would instead use `None` as default value, and it just means *default*. It does NOT imply a boolean evaluation to make it a False.

And the actual default behavior will and only will be available via documentation. See line 419 in this PR.

Reply:
Do you mean the caller (me :)) can specify None as the input value of instance_discovery to say "use default"?

Reply:
Yes :-)